### PR TITLE
K8s: add plumbing for e2e tests using grafana/e2e

### PR DESCRIPTION
**Pull request by charandas**
**What is this feature?**

This will allow running e2e tests after docker image builds. We are using something similar in enterprise and using `grafana/e2e` go pkg. Its working fairly well.

**Why do we need this feature?**

Using `grafana/e2e` uncouples us from Drone and allows us to launch arbitrary number of docker services all from within go code. This may prove relevant when cutting over to GitHub actions in the future.

**Who is this feature for?**

Grafana App Platform

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: scripts/drone/steps/lib.star [hunk 1]
```diff
@@ -1021,6 +1021,57 @@ def integration_benchmarks_step(name, environment = None):
...
+    cmds = [
+        'apk add --update --no-cache build-base make git coreutils bash docker gcc libc-dev',
+        'export GRAFANA_IMAGE=grafana/grafana-enterprise-dev:$(cat $OSS_BUILD_VERSION_FILE)',
+        # tag the image to a arch-agnostic name (publish step does this in a better way, but we don't want to wait on it)
+        'docker tag grafana/grafana-enterprise-image-tags:$(cat $OSS_BUILD_VERSION_FILE)-amd64 $$GRAFANA_IMAGE',
+        'export E2E_NETWORK_NAME=e2e-$DRONE_BUILD_NUMBER-$DRONE_STAGE_NUMBER-$DRONE_STEP_NUMBER',
+        'mkdir -p $HOME/.docker',
+        'printenv GCR_CREDS > $HOME/.docker/config.json',
+        "go clean -testcache",
+        "go test -p=1 -count=1 -covermode=atomic -timeout=5m ./pkg/apiserver/apiserver_test/...",
```
Comment (charandas) [inline scripts/drone/steps/lib.star:1034]: placeholder test package

File: scripts/drone/steps/lib.star [hunk 2]
```diff
@@ -1021,6 +1021,57 @@ def integration_benchmarks_step(name, environment = None):
 
     return integration_tests_steps("{}-benchmark".format(name), cmds, environment = environment)
 
+def e2e_apiserver_tests_step():
+    cmds = [
+        'apk add --update --no-cache build-base make git coreutils bash docker gcc libc-dev',
+        'export GRAFANA_IMAGE=grafana/grafana-enterprise-dev:$(cat $OSS_BUILD_VERSION_FILE)',
```
Comment (toddtreece) [inline scripts/drone/steps/lib.star:1027]: is `grafana-enterprise-dev` correct for OSS?

File: .drone.yml [hunk 3]
```diff
@@ -866,6 +866,39 @@ steps:
...
+  image: grafana/build-container:1.7.7
+  name: export-version
+- commands:
+  - apk add --update --no-cache build-base make git coreutils bash docker gcc libc-dev
+  - export GRAFANA_IMAGE=grafana/grafana-enterprise-dev:$(cat $OSS_BUILD_VERSION_FILE)
+  - docker tag grafana/grafana-enterprise-image-tags:$(cat $OSS_BUILD_VERSION_FILE)-amd64
+    $$GRAFANA_IMAGE
+  - export E2E_NETWORK_NAME=e2e-$DRONE_BUILD_NUMBER-$DRONE_STAGE_NUMBER-$DRONE_STEP_NUMBER
+  - mkdir -p $HOME/.docker
+  - printenv GCR_CREDS > $HOME/.docker/config.json
```
Comment (toddtreece) [inline .drone.yml:882]: is there a risk of leaking these creds? i'm not sure if this is useful, but it looks like [this is how it's handled](https://github.com/grafana/grafana/blob/ec7a2476740f31ae02ca438e73b7767902dab64b/scripts/drone/events/cron.star#L26) in another pipeline

**Comment by charandas** [comment]
Closing since tests can be covered in grafana-enterprise because of https://github.com/grafana/grafana/pull/93799

File: scripts/drone/steps/lib.star (hunk 2 above)
Comment (charandas) [reply scripts/drone/steps/lib.star:1027]: this was a typo, good catch

File: .drone.yml (hunk 3 above)
Comment (charandas) [reply .drone.yml:882]: good to know. will weigh this in on the other side (in enterprise)

**Pull request closed without merging**